### DISCUSSION TITLE
pdm build isn't packaging the source files

### DISCUSSION THREAD

**Discussion by tvdboom** [pdm-project/pdm#1534] [Q&A] [answered]
Hi, I have a package called [atom](https://github.com/tvdboom/ATOM) (pypi name is atom-ml). The source files are in the directory `atom`. When I use pdm build, the `.tar.gz` file is build correctly but it doesn't include the source files in `atom`. 

The (simplified) `pyproject.toml looks` like this:
```
[build-system]
requires = ["pdm-pep517"]
build-backend = "pdm.pep517.api"

[project]
name = "atom-ml"
description = "A Python package for fast exploration of machine learning pipelines"
authors = [{name = "Mavs", email = "m.524687@gmail.com"}]
license = {file = "LICENSE"}
readme = "README.md"
requires-python = ">=3.8"
keywords = ["Python package", "Machine Learning", "Modelling", "Data Pipeline"]
dynamic = ["version"]
classifiers = [
    "Programming Language :: Python :: 3.8",
    "Programming Language :: Python :: 3.9",
    "Programming Language :: Python :: 3.10",
    "Topic :: Scientific/Engineering :: Artificial Intelligence",
    "License :: OSI Approved :: MIT License",
    "Operating System :: OS Independent",
]

[tool.pdm.version]
source = "file"
path = "atom/utils.py"

[tool.pdm.build]
package-dir = "atom"
includes = ["README.md", "LICENSE", "images/"]
source-includes = ["tests/*.py"]
```

The resulting .tar contains:
![image](https://user-images.githubusercontent.com/32366550/204486390-84d617bb-2a48-4d4a-9482-af408096ccb7.png)

What am I doing wrong? Obviously, the library isn't installed when using `python -m pip install . -vv`.

Thanks in advance.

**Comment by frostming** [comment, marked as answer]
From the [doc](https://pdm.fming.dev/dev/pyproject/build/#include-and-exclude-files):

> When using `includes` the default includes will be overridden. You have to add the package paths manually.

You mistake the `includes` as **add** while instead it should be **override**

**Reply by tvdboom** [reply]
Thanks for the quick reply. I changed that part of the `pyproject.toml` file to this and that fixed it.
```
[tool.pdm.build]
package-dir = "atom"
source-includes = ["tests/*.py"]
```

I immediately have a follow-up question though. After installing the package and trying to import it I get this error
![image](https://user-images.githubusercontent.com/32366550/204505244-8f5f9cdb-bf8b-4e46-a046-13c26542dc1c.png)

I guess the issue is that the package has the same name as one of the files in the directory `atom.py`, but I don't want to rename the file. How can I solve this? I didn't had this issue when installing with a `setup.py` script.

**Reply by tvdboom** [reply]
I found the problem. Apparently, when installing, it doesn't add the directory `atom` to site-packages but every .py file inside `atom` individually. Not sure why.

**Reply by frostming** [reply]
> package-dir = "atom"

This is wrong, package dir is the **parent directory** to find package from, not the package itself, just change it to `includes = ["atom"]`

**Reply by tvdboom** [reply]
Yes, that fixed it. Thanks again
